megamek/src/megamek/common/Crew.java

```diff
@@ -1488,6 +1488,30 @@ public String getExtraDataValue(int crewIndex, String key) {
             return extraData.get(crewIndex).get(key);
         }
     }
+
+    public String writeExtraDataToXMLLine(int pos) {
+        Map<String, String> dataRow = getExtraDataForCrewMember(pos);
+
+        if (dataRow != null) {
+            StringBuilder output = new StringBuilder();
+
+            output.append("\" extraData=\"");
+
+            String[] keySet = new String[dataRow.keySet().size()];
+            dataRow.keySet().toArray(keySet);
+
+            for (int k = 0; k < keySet.length; k++) {
+                output.append(keySet[k]).append(",").append(dataRow.get(keySet[k]));
+                if (k < keySet.length - 1) {
+                    output.append(",");
+                }
+            }
```

Comment:
Why not a `for` over the `entrySet`? This is pretty expensive in terms of allocations.

Reply:
Easier parsing on load, and... This isn't that bad in terms of allocations?
EntrySet is a harder entity to parse on load and is more limiting to potential characters in the set.

Reply:
I mean:
```
boolean first = false;
for (EntrySet<K, V> row : dataRow.entrySet()) {
  if (!first) {
    output.append(",");
  } else {
    first = false;
  }
  output.append(row.getKey()).append(",").append(row.getValue());
}
```

Reply:
Also, would it be more natural to split these int pairs by `|` and key-values by `=`? This way reading the file is obvious as to the intent.

Side question: is `,`, `|`, or `=` valid in a key or value?